src/driver-sinowealth.c

```diff
@@ -520,7 +824,10 @@ sinowealth_get_active_profile(struct ratbag_device *device)
 static int
 sinowealth_set_active_profile(struct ratbag_device *device, unsigned int index)
 {
-	assert(index <= SINOWEALTH_NUM_PROFILES - 1);
+	if (index > SINOWEALTH_NUM_PROFILES - 1) {
```

Comment:
We can't because this driver doesn't store profile count in the `drv_data`. This is because in contrast to hidpp devices, all Sinowealth-based mice have 2 profiles, not more, not less. 

I only enabled one, though, as by default second profile has broken configuration data, which would need to get reset in such cases.
(..which technically makes the whole profile selecting code bloat as it's gonna be used only if the source code is altered.) :)

Reply:
I am sorry, I meant to write `if (index >= SINOWEALTH_NUM_PROFILES) {`
Sorry for the confusion, but feel free to ignore it. it's just nitpicking

Reply:
This looks better, thanks.